src/utils/test.js

```diff
@@ -19,17 +19,28 @@ module.exports = {
       },
       message,
 
+      /**
+       * Get the action handler that the bot registers.
+       * @param {Number} index Optional. The index of the handler to fetch. For
+       *    Bots that register multiple handlers. This is in the order that the
+       *    handlers are registered. Defaults to 0, the first handler.
+       */
+      getActionHandler: (index = 0) => {
+        if (action.mock.calls.length > index) {
+          return action.mock.calls[index].slice(-1).pop();
+        }
+        return null;
+      },
+
       /**
        * Get the handler that the bot registers.
        * @param {Number} index Optional. The index of the handler to fetch. For
        *    Bots that register multiple handlers. This is in the order that the
        *    handlers are registered. Defaults to 0, the first handler.
        */
       getHandler: (index = 0) => {
-        if (message.mock.calls.length > 0) {
-          return message.mock.calls[index][
-            message.mock.calls[index].length - 1
-          ];
+        if (message.mock.calls.length > index) {
```

Comment:
I don't know Javascript patterns, but since this is an exact copy of the preceding change, is it worth making a helper "safe access" method that does this check and slice?

Reply:
Probably yes. I felt squicky about the repetition but decided to leave it alone for now and embrace my inner [Rule of Three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)).